src/MCPServer/lib/assets/workflow-schema-v1.json

```diff
@@ -0,0 +1,444 @@
+            { "$ref": "#/definitions/link_model_GetServiceOutput" },
+            { "$ref": "#/definitions/link_model_GetServiceOutputPrompt" },
+            { "$ref": "#/definitions/link_model_GetUnitVar" },
+            { "$ref": "#/definitions/link_model_SetUnitVar" }
+          ]
+        },
+        "exit_codes": {
+          "type": "object",
+          "minProperties": 0,
+          "additionalProperties": {
+            "type": "object",
+            "properties": {
+              "job_status": {
+                "oneOf": [
+                  { "$ref": "#/definitions/job_status" },
+                  { "type": "null" }
+                ]
+              },
+              "link_id": {
+                "oneOf": [
+                  { "$ref": "#/definitions/uuid" },
+                  { "type": "null" }
+                ]
+              }
+            }
+          }
+        },
+        "fallback_job_status": {
+          "oneOf": [
+            { "$ref": "#/definitions/job_status" },
+            { "type": "null" }
+          ]
+        },
+        "fallback_link_id": {
+          "oneOf": [
+            { "$ref": "#/definitions/uuid" },
+            { "type": "null" }
+          ]
+        },
+        "description": {
+          "$ref": "#/definitions/translations"
+        },
+        "group": {
+          "$ref": "#/definitions/translations"
+        }
+      },
+      "additionalProperties": false,
+      "required": [
+        "config",
+        "exit_codes",
+        "fallback_job_status",
+        "fallback_link_id",
+        "description",
+        "group"
+      ]
+    },
+    "watched_dir": {
+      "type": "object",
+      "properties": {
+        "chain_id": {
+          "$ref": "#/definitions/uuid"
+        },
+        "only_dirs": {
+          "type": "boolean"
+        },
+        "path": {
+          "type": "string"
+        },
+        "unit_type": {
+          "type": "string",
+          "enum": [
+            "Transfer",
+            "DIP",
+            "SIP"
+          ]
+        }
+      },
+      "additionalProperties": false,
+      "requried": [
+        "chain_id",
+        "only_dirs",
+        "path",
+        "unit_type"
+      ]
+    },
+    "link_model_StandardTaskFile": {
+      "type": "object",
+      "properties": {
+        "@manager": {"type": "string", "pattern": "linkTaskManagerFiles"},
+        "@model": {"type": "string", "pattern": "StandardTaskConfig"},
+        "execute": {
+          "type": "string"
+        },
+        "arguments": {
+          "type": ["string", "null"]
+        },
+        "filter_file_start": {
+          "type": ["string", "null"]
+        },
+        "filter_file_end": {
+          "type": ["string", "null"]
+        },
+        "filter_subdir": {
+          "type": ["string", "null"]
+        },
+        "stdout_file": {
+          "type": ["string", "null"]
+        },
+        "stderr_file": {
+          "type": ["string", "null"]
```

Comment:
I should have provided more context with this conversation. Null values take up a lot of space in the `workflow.json` when the schema can dictate what is there, and what can be implied to be null. It would be good to consider making a change that can reduce null values in the `json` itself in future.

Reply:
Ok, I've included your document in https://gist.github.com/sevein/63e751acd37bde5590d680344ad8b3d5.